utt/cmd_report.py

```diff
@@ -56,54 +82,27 @@ def execute(args):
 ]
 
 
-def _activities_from_entries(entries_grouped_by_day):
-    activities_grouped_by_day = collections.defaultdict(list)
-
-    for date, entries in entries_grouped_by_day.items():
-        activities_grouped_by_day[date] = list(
-            _activities_from_entries_day(entries))
-
-    return activities_grouped_by_day
-
+def _collect_activities(start_date, end_date, entries):
+    start_datetime = datetime.datetime(start_date.year, start_date.month,
+                                       start_date.day)
+    end_datetime = datetime.datetime(end_date.year, end_date.month,
+                                     end_date.day, 23, 59, 59, 99999)
+    activities = []
+    for entry_pair in _pairwise(entries):
+        activity = Activity(entry_pair[0].datetime, entry_pair[1]).clip(
+            start_datetime, end_datetime)
+        if (activity.duration > datetime.timedelta() and
+                activity.name.name != HELLO):
+            activities.append(activity)
+    return sorted(activities, key=lambda act: act.start)
 
-def _activities_from_entries_day(entries):
-    return (Activity(entry_pair[0].datetime, entry_pair[1])
-            for entry_pair in _pairwise(entries))
 
-
-def _add_current_entry(report_date, entries, now, current_activity_name,
+def _add_current_entry(entries, now, current_activity_name,
                        disable_current_activity):
-    today = now.date()
-    if report_date == today and entries and entries[-1].datetime < now and \
-            not disable_current_activity:
+    if entries and entries[-1].datetime < now and not disable_current_activity:
```

Comment:
The current activity should only be added if the user looks at today's report. By removing `report_date == today` the current activity is now added even when the user look at, say, yesterday's report. This creates a bogus current activity:

```
$ cat /root/.local/share/utt/utt.log   
2018-08-18 22:00 hello
2018-08-18 22:10 foo
```

```
$ utt --now "2018-08-19 23:00" report 2018-08-18

----------------------- Saturday, Aug 18, 2018 (week 33) -----------------------

Working Time: 1 day, 1h00 (0h10 + 1 day, 0h50) [25h00]
Break   Time: 0h00 [0h00]

----------------------------------- Projects -----------------------------------

(1 day, 1h00) : -- Current Activity --, foo

---------------------------------- Activities ----------------------------------

(1 day, 0h50) : -- Current Activity --
(0h10) : foo


----------------------------------- Details ------------------------------------

(0h10) 22:00-22:10 foo
(1 day, 0h50) 22:10-23:00 -- Current Activity --
```